spec/models/category_spec.rb

```diff
@@ -61,6 +61,32 @@
       end
     end
 
+    context 'for pileid-source Category' do
+      let(:category) { create(:category, name: 28301, source: 'pileid') }
+      let(:course) { create(:course, id: 28301) }
+      let!(:article) { create(:article, title: 'America') }
+
+      it 'updates article titles for categories associated with courses' do
+        expect(described_class.last.article_titles).to be_empty
+
+        VCR.use_cassette 'categories' do
+          described_class.refresh_categories_for(Course.find_by(id: 28301))
+          expect(described_class.last.article_titles).not_to be_empty
+          expect(described_class.last.article_ids).to include(article.id)
+        end
+      end
+
+      it 'return empty list' do
```

Comment:
I'm unclear what is being tested here. It looks like Course 1234 doesn't exist during the test.